list/provisioning.cattle.io.cluster.vue

```diff
@@ -5,22 +5,30 @@ import { allHash } from '@/utils/promise';
 import { CAPI, MANAGEMENT } from '@/config/types';
 import { MODE, _IMPORT } from '@/config/query-params';
 
+function addIfHave(store, hash, hashProperty, resourceType) {
+  if ( store.getters['management/schemaFor'](resourceType) ) {
+    hash[hashProperty] = store.dispatch('management/findAll', { type: resourceType });
+  }
+}
+
 export default {
   components: { ResourceTable, Masthead },
 
   async fetch() {
-    const hash = await allHash({
+    const hash = {
       mgmtClusters:       this.$store.dispatch('management/findAll', { type: MANAGEMENT.CLUSTER }),
-      // Used to determine non-rke2 node counts
-      mgmtNodes:          this.$store.dispatch('management/findAll', { type: MANAGEMENT.NODE }),
-      mgmtPools:          this.$store.dispatch('management/findAll', { type: MANAGEMENT.NODE_POOL }),
-      mgmtTemplates:      this.$store.dispatch('management/findAll', { type: MANAGEMENT.NODE_TEMPLATE }),
       rancherClusters:    this.$store.dispatch('management/findAll', { type: CAPI.RANCHER_CLUSTER }),
-      machineDeployments: this.$store.dispatch('management/findAll', { type: CAPI.MACHINE_DEPLOYMENT })
-    });
+    };
+
+    addIfHave(this.$store, hash, 'mgmtNodes', MANAGEMENT.NODE);
```

Comment:
If you define addIfHave() inside fetch() it would have access to `this.$store` and `hash` without passing around.

Reply:
Difference in styling - verbose vs resources consumption. Currently more params but the function wouldn't be recreated every time fetch is called. I've gone off-piste with this, and fetch won't ever be called crazy often at all. Will fix.

Reply:
Moved to #3720 (where it's been resolved)